papers/joss/paper.md

```diff
@@ -21,9 +21,12 @@ bibliography: paper.bib
 
-In the domain of NLP, there is an analogous set of transformations that have been shown to increase model accuracy, or make models more robust in the face of adversarial inputs. These include adding superfluous whitespace, swapping an individual word for a synonym, and appending the word "love" to the end of a sentence (this last one, in an attempt to fool sentiment classifiers) [@zhang2015characterlevel;@coulombe2018text;@grndahl2018need]. There are a small number of libraries that contain text augmentation functions, for example [noisemix](https://github.com/noisemix/noisemix) and [EDA](https://github.com/jasonwei20/eda_nlp) [@wei2019eda], but these tend to be associated with conference papers and have little documentation, or only implement a small subset of common transforms.
+In the domain of NLP, there is an analogous set of transformations that have been shown to increase model accuracy. Common transformations include adding superfluous whitespace, swapping an individual word for a synonym, replacing a word with a misspelled version, and round-tripping a sentence through a translation model into another language and back again [@zhang2015characterlevel;@coulombe2018text]. Some transforms, like misspellings or synonyms, attempt to directly model an unobserved but common change in the input features. Others, like adding a space in the middle of a word, act similarly to dropout for word-level models, in that they effectively the word as an input (e.g. the two halves of the word are unlikely to be present in the vocabulary).
 
-Here, we introduce niacin, a python library for performing a large set of commonly-used text transformations on a given input, with some probability. For example, a text augmentation technique used in technologies like word2vec [@mikolov2013distributed] and Glove [@pennington2014glove] is to produce negative examples (bad or wrong inputs) by replacing words with ones that are too specific for the context of the rest of the sentence. In niacin, you can use a function called `add_hyponyms` for this, which, at a replacement probability of 1.0, produces outputs like this:
+
+There are also sets of transforms whose aim is to make models more robust in the face of adversarial inputs. These transforms include replacing letters with symbols that are visually similar, adding nuisance characters around words, adding random characters at the end of the document, and appending the word "love" to the end of a sentence [@grndahl2018need]. Adding nuisance characters around words, or replacing characters with symbols, disrupt the bahvior of vocabulary-based models, in that they will not know how to encode an input like "sch00l", although it is visually clear to an experienced reader of English that this is the word "school". Adding words with very positive or very negative connotations have been shown to be an effective way to disrupt sentiment classifiers, which often rely on counting the number of times such words appear.
```

Comment:
Typo in `bahvior`, or was it intentional ;)

Reply:
Lol it might feel a bit bah